File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        
        int maxi=-1;
        int n=heights.size();
        int left=0;
        int right=n-1;

        while(left<right){
          
          int h=min(heights[left],heights[right]);
          
          int b=right-left;
          
          int area=h*b;
          
          if(heights[left]>heights[right]) right--;
          else  left++;
          
          
          maxi=max(maxi,area);



        }
        return maxi;

    }
};

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-2.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        
        n=len(arr)
        ans=[0]*n

        for i in range (n):

            right=-1

            for j in range(i+1,n):

                right=max(right,arr[j])

            ans[i]=right

        return ans     